dart/test/utils_test.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test/test.dart';
+import 'package:vm_service_lib/utils.dart';
+
+@TestOn('vm')
+void main() {
+  group('getVmWsUriFromObservatoryUri', () {
+    void check(String input, String expected) {
+      final inputUri = Uri.parse(input);
+      final actualUri = getVmWsUriFromObservatoryUri(inputUri);
+      expect(actualUri.toString(), equals(expected));
+    }
+
+    test('handles http URIs',
+        () => check('http://localhost:123/', 'ws://localhost:123/ws'));
```

Comment:
nit: this formatting is a bit odd looking. I would suggest using () {
  check('http://localhost:123/', 'wss://localhost:123/ws');
}

No need to write this code but if you have a similar situation in the future it might make sense to have a 
Map<String, String> of input to expected uris and then have a single test case that
iterates through all key-value pairs in the map expecting that the uris match.

Reply:
I started it that way, but wasn't sure if having only a single test name without descriptions was good. However, it did turn out a mess, so I've changed it to a `Map<String, String>`. PTAL!